usaspending_api/download/management/commands/download_sqs_worker.py

```diff
@@ -1,185 +1,154 @@
+                #   - the zip will use 'w' write mode to create from scratch each time
+                # The worker function controls the maximum allowed runtime of the job
 
-def create_and_start_new_process(download_job_id):
-    download_app = Process(
-        name="Download Service Worker Proccess", target=download_service_app, args=(download_job_id,)
-    )
-    download_app.start()
+            except (QueueWorkerProcessError, QueueWorkDispatcherError) as exc:
+                _handle_queue_error(exc)
+
+            # When you receive an empty response from the queue, wait before trying again
+            if not message_found:
+                time.sleep(1)
 
-    return download_app
+            # If this process is exiting, don't poll for more work
+            keep_polling = not dispatcher.is_exiting
 
 
 def download_service_app(download_job_id):
-    download_job = retrieve_download_job_from_db(download_job_id)
-    write_to_log(message="Starting new Download Service App with pid {}".format(os.getpid()), download_job=download_job)
+    download_job = _retrieve_download_job_from_db(download_job_id)
+    download_job_details = download_job_to_log_dict(download_job)
+    log_job_message(
+        logger=logger,
+        message="Starting processing of download request",
+        job_type=JOB_TYPE,
+        job_id=download_job_id,
+        other_params=download_job_details,
+    )
+    generate_download(download_job=download_job)
 
-    # Retrieve the data and write to the data files
-    try:
-        generate_download(download_job=download_job)
-    except Exception:
-        write_to_log(message="Caught exception", download_job=download_job, is_error=True)
-        return 11  # arbitrary positive integer
 
-    return 0
+def _retrieve_download_job_from_db(download_job_id):
+    download_job = DownloadJob.objects.filter(download_job_id=download_job_id).first()
 
+    if download_job is None:
+        raise DownloadJobNoneError(download_job_id)
 
-def remove_message_from_sqs(message):
-    if message is None:
-        return
-    try:
-        message.delete()
-    except Exception:
-        write_to_log(message="Unable to delete SQS message. Message might have previously been released or removed")
+    return download_job
 
 
-def set_sqs_message_visibility(message, new_visibility):
-    if message is None:
-        write_to_log(message="No SQS message to modify. Message might have previously been released or removed")
-        return
-    try:
-        message.change_visibility(VisibilityTimeout=new_visibility)
-    except botocore.exceptions.ClientError:
-        write_to_log(message="Unable to set VisibilityTimeout. Message might have previously been released or removed")
+def _update_download_job_status(download_job_id, status, error_message=None, overwrite_error_message=False):
+    """Handles status updates on the DownloadJob records
 
+    Args:
+        download_job_id (int): id of the DownloadJob record being processed
```

Comment:
No reason to change now. If the parameters types are useful to document then adding type hints are the recommended documentation